File: frontend/public/reportes.php
<?php
declare(strict_types=1);
require __DIR__ . '/../includes/bootstrap.php';

exigir_sesion(['ANALISTA', 'ADMIN']);
$respuesta = api('GET', '/api/reportes/resumen');
$resumen = $respuesta['estado'] === 200 ? $respuesta['datos'] : null;
$errorGeneral = $resumen === null ? errores_de($respuesta)[0] : null;
$total = $resumen ? max(1, (int)$resumen['total_solicitudes']) : 1;

encabezado('Reportes');
?>
<section class="panel">
    <h1>Resumen de solicitudes</h1>
    <?= $errorGeneral ? aviso_error($errorGeneral) : '' ?>
    <?php if ($resumen): ?>
        <div class="cifras">
            <p><strong><?= e($resumen['total_solicitudes']) ?></strong> solicitudes registradas</p>
            <p><strong><?= e(quetzales($resumen['monto_preaprobado'])) ?></strong> en créditos preaprobados o aceptados</p>
        </div>
        <h2>Por estado</h2>
        <ul class="barras">
            <?php foreach ($resumen['por_estado'] as $fila): ?>
                <li>
                    <span class="barras__rotulo"><?= e(texto_estado((string)$fila['estado'])) ?></span>
                    <span class="barras__pista"><span class="barras__valor" style="width: <?= e(round($fila['cantidad'] * 100 / $total)) ?>%"></span></span>
                    <span class="barras__numero"><?= e($fila['cantidad']) ?></span>
                </li>
            <?php endforeach; ?>
        </ul>
        <h2>Por producto</h2>
        <div class="tabla-desplazable">
            <table class="tabla">
                <thead><tr><th>Producto</th><th class="num">Solicitudes</th><th class="num">Monto solicitado</th></tr></thead>
                <tbody>
                <?php foreach ($resumen['por_producto'] as $fila): ?>
                    <tr>
                        <td><?= e($fila['producto_codigo']) ?></td>
                        <td class="num"><?= e($fila['cantidad']) ?></td>
                        <td class="num"><?= e(quetzales($fila['monto_total'])) ?></td>
                    </tr>
                <?php endforeach; ?>
                </tbody>
            </table>
        </div>
    <?php endif; ?>
</section>
<?php pie(); ?>

File: backend/app/rutas/reportes.py
"""Reporte resumen de solicitudes para analistas y administradores."""
from flask import Blueprint, jsonify

from ..db import consultar
from ..seguridad import requiere_rol

bp = Blueprint("reportes", __name__, url_prefix="/api/reportes")


@bp.get("/resumen")
@requiere_rol("ANALISTA", "ADMIN")
def resumen():
    por_estado = consultar(
        "SELECT estado, COUNT(*) AS cantidad FROM solicitudes GROUP BY estado ORDER BY estado"
    )
    por_producto = consultar(
        "SELECT producto_codigo, COUNT(*) AS cantidad, COALESCE(SUM(monto), 0) AS monto_total"
        " FROM solicitudes GROUP BY producto_codigo ORDER BY producto_codigo"
    )
    preaprobado = consultar(
        "SELECT COALESCE(SUM(monto), 0) AS total FROM solicitudes"
        " WHERE estado IN ('PREAPROBADA', 'ACEPTADA')",
        uno=True,
    )
    total = sum(fila["cantidad"] for fila in por_estado)
    return jsonify(
        total_solicitudes=total,
        por_estado=por_estado,
        por_producto=por_producto,
        monto_preaprobado=preaprobado["total"],
    )
